visitors/http_rest/spec_visitor_default.go

```diff
@@ -0,0 +1,315 @@
+		}
+	case *pb.Optional_None:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "None", node.None) // TODO: don't recurse?
+		}
+	case *pb.Optional_Data:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "Data", node.Data)
+		}
+	case *pb.OneOf:
+		if node != nil {
+			keepGoing = visitStructMembers(ctx, vm, m, "Options", node.Options)
+		}
+	// These all have No non-basic types as children, so no further recursion is needed
+	case *pb.MethodID: // FIXME: should we even enter this one? Not used by SpecVisitor
+	case *pb.HTTPMethodMeta:
+	case *pb.HTTPPath:
+	case *pb.HTTPQuery:
+	case *pb.HTTPHeader:
+	case *pb.HTTPCookie:
+	case *pb.HTTPBody:
+	case *pb.HTTPEmpty:
+	case *pb.HTTPAuth:
+	case *pb.HTTPMultipart:
+	case *pb.FormatOption_StringFormat: // only contains a string
+	case *pb.None:
+	default:
+		// Fallback to reflection-based implementation
+		return go_ast.DefaultVisitChildren(ctx, vm, m)
+	}
+	return keepGoing
+}
+
+// Visit the struct members given by <field1> <value1> <field2> <value2>
+func visitStructMembers(ctx Context, vm VisitorManager, inStruct interface{}, fields ...interface{}) Cont {
+	keepGoing := Continue
+	for i := 0; i < len(fields); i += 2 {
+		field := fields[i].(string)
+		value := fields[i+1]
+		// Recurse into the member
+		keepGoing := VisitInterface(ctx.EnterStruct(inStruct, field), vm, value)
+
+		// Validate return value
+		switch keepGoing {
+		case Abort, Stop:
+			return keepGoing
+		case Continue:
+			// fall through and get the next value, if any
+		case SkipChildren:
+			panic("VisitInterface returned SkipChildren")
+		default:
+			panic(fmt.Sprintf("Unknown Cont value: %d", keepGoing))
+		}
+	}
+	return keepGoing
+}
+
+func visitSliceMember(ctx Context, vm VisitorManager, inSlice interface{}, index int, value interface{}) Cont {
+	// Recurse into the member
+	keepGoing := VisitInterface(ctx.EnterArray(inSlice, index), vm, value)
+
+	// Validate return value
+	switch keepGoing {
+	case Abort, Stop, Continue:
+		return keepGoing
+	case SkipChildren:
+		panic("VisitInterface returned SkipChildren")
+	default:
+		panic(fmt.Sprintf("Unknown Cont value: %d", keepGoing))
+	}
+}
+
+func visitMapMember(ctx Context, vm VisitorManager, inSlice interface{}, key string, value interface{}) Cont {
+	// Recurse into the member
+	keepGoing := VisitInterface(ctx.EnterMapValue(inSlice, key), vm, value)
+
+	// Validate return value
+	switch keepGoing {
+	case Abort, Stop, Continue:
+		return keepGoing
+	case SkipChildren:
+		panic("VisitInterface returned SkipChildren")
+	default:
+		panic(fmt.Sprintf("Unknown Cont value: %d", keepGoing))
+	}
+}
+
+// Visits the node m of any type, whose context is c. This should never return SkipChildren.
+// Implements all the logic to enter, visit children, and leave, while obeying the visitor's returned
+// Cont value.
+// This is a copy of astVisitor's visit function.
+func VisitInterface(c Context, vm VisitorManager, m interface{}) Cont {
```

Comment:
As the comment says, this is basically just `visit()` lifted to the `VisitorManager` level. I never really understood why we need both visitors and `VisitorManager`. Seems like we should eventually just merge the two together.

Reply:
Deduplicated as part of #94.